pkg/controller/clusters/status.go

```diff
@@ -41,10 +42,11 @@ func UpdateClusterStatusForWorkers(ctx context.Context, client client.Client, cl
 
 // UpdateClusterStatusForCNI updates the Cluster status for the default cni before the control plane is ready. The CNI reconciler
 // handles the rest of the logic for determining the condition and updating the status based on the current state of the cluster.
-func UpdateClusterStatusForCNI(ctx context.Context, cluster *anywherev1.Cluster) {
+func UpdateClusterStatusForCNI(ctx context.Context, client client.Client, cluster *anywherev1.Cluster) error {
```

Comment:
There is starting to be too much duplication between this and the cni reconciler. plus we keep bleeding cilium specific logic here.

What do you think about adding a new method to the CNI reconciler interface like  `UpdateClusterStatusForCNI`. And then just call that method from the cluster reconciler where you update the status. That way there is only one way to update the status, regardless of the type of cluster and the scenario.

Reply:
Sounds good. I'll move this method from here and add a new method to the CNI reconciler interface to avoid bleeding cilium specific logic here.

Can you clarify what your thoughts on handling the duplication is exactly? The self managed clusters don't use the cni reconciler and the status updates in the cni reconciler are integrated into the reconcile process. So it would be hard to remove duplication (even after the move), at least without changing that.

Reply:
can't this new method set the information for both workload and management clusters?